precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/BackStackManager.kt

```diff
@@ -116,9 +116,9 @@ internal class BackStackManager : LifecycleObserver {
                 } else 0
             }
             if (index != -1) {
-                val stacks = currentBackStacks.subList(
+                val stacks = backStacks.value.subList(
                     if (popUpTo.inclusive) index else index + 1,
-                    currentBackStacks.size
+                    backStacks.value.size - 1
```

Comment:
When the destination uses `singleLaunchTop`, the backstack order will change, and also the size will stay the same, so sublisting from the `currentBackStacks` is the root cause of the crash, I think that by using the updated `backstacks` it should work for all cases, but let me know what you think.

Reply:
Thanks for the pr, can you add some unit test for this use case?

Reply:
Done in fcebb58